### PR TITLE
nametags v0

### LINKED ISSUE #80: draw player nametags

### PR DESCRIPTION
**Pull request by zpl-zak**
v0 impl, closes #80

### CODE REVIEW

File: code/client/src/core/modules/human.cpp [hunk 1]
```diff
@@ -122,6 +126,31 @@ namespace MafiaMP::Core::Modules {
...
+                    float distFromCam                    = headPos.dist(camPos);
+                    if (distFromCam <= 250.0f /* @todo: make configurable */) {
+                        headPos.z += 0.335f + (distFromCam * 0.03f);
+
+                        SDK::ue::sys::math::C_Vector2 screenPos;
+                        bool onScreen = false;
+                        float unkFloat1, unkFloat2;
+                        camera->GetScreenPos(screenPos, headPos, onScreen, &unkFloat1, &unkFloat2, true);
+                        if (onScreen) {
+                            Framework::Logging::GetLogger("aha")->debug("nick hp: {}", hp);
```
Comment (Segfaultd) [inline code/client/src/core/modules/human.cpp:149]: Oopsi

File: code/client/src/core/modules/human.cpp [hunk 2]
```diff
@@ -122,6 +126,31 @@ namespace MafiaMP::Core::Modules {
...
+
+                const auto humanPtr = tracking.human;
+                const auto hp       = updateData->_healthPercent;
+                const auto nickname = humanData->nickname;
+                gApplication->GetImGUI()->PushWidget([humanPtr, hp, nickname]() {
+                    const auto displaySize               = ImGui::GetIO().DisplaySize;
+                    auto gameCamera                      = SDK::ue::game::camera::C_GameCamera::GetInstanceInternal();
+                    auto camera                          = gameCamera->GetCamera(SDK::ue::game::camera::E_GameCameraID::CAMERA_PLAYER_MAIN_1);
+                    auto camPos                          = camera->GetPos();
+                    static const auto headBoneHash       = SDK::ue::sys::utils::C_HashName::ComputeHash("Head");
```
Comment (Segfaultd) [inline code/client/src/core/modules/human.cpp:138]: Why static?

File: code/client/src/core/modules/human.cpp [hunk 3]
```diff
@@ -122,6 +126,31 @@ namespace MafiaMP::Core::Modules {
                         tracking.human->SetRot({newRot.x, newRot.y, newRot.z, newRot.w});
                     }
                 }
+
+                const auto humanPtr = tracking.human;
+                const auto hp       = updateData->_healthPercent;
+                const auto nickname = humanData->nickname;
+                gApplication->GetImGUI()->PushWidget([humanPtr, hp, nickname]() {
+                    const auto displaySize               = ImGui::GetIO().DisplaySize;
+                    auto gameCamera                      = SDK::ue::game::camera::C_GameCamera::GetInstanceInternal();
```
Comment (Segfaultd) [inline code/client/src/core/modules/human.cpp:135]: Make sure you have a valid pointer

File: code/client/src/core/modules/human.cpp [hunk 4]
```diff
@@ -122,6 +126,31 @@ namespace MafiaMP::Core::Modules {
...
+                    auto camera                          = gameCamera->GetCamera(SDK::ue::game::camera::E_GameCameraID::CAMERA_PLAYER_MAIN_1);
```
Comment (Segfaultd) [inline code/client/src/core/modules/human.cpp:136]: Make sure you have a valid pointer

**Review by Segfaultd — CHANGES_REQUESTED** [review]

File: code/client/src/core/modules/human.cpp (hunk 2 above)
Comment (zpl-zak) [reply code/client/src/core/modules/human.cpp:146]: there's no difference, still has to be static.

Comment (zpl-zak) [reply code/client/src/core/modules/human.cpp:146]: putting it outside lambda means extra code to capture static var, which is not required.

**Pull request merged**